### ISSUE TITLE
Empy commit message being passed to `git`

### ISSUE THREAD

**Issue by martinbaillie** [douo/magit-gptcommit#13]
Thanks for the neat package. I was looking for something just like this 👍🏻 

I've just had a quick play (latest version) and I'm not sure if its PEBKAC or what, but despite a `COMMIT_MSG` buffer looking great, complete with generated content, a typical <kbd>C-C</kbd><kbd>C-c</kbd> to commit in Magit is resulting in "Aborting commit due to empty commit message".

This is regardless of the way I have generated the message using `magit-gptcommit`. 

If I override `magit-gptcommit-commit-accept` to add `(save-buffer)` at the end then it works but that feels heavyweight and that it might mess up some other stuff. I naively tried things like `(set-buffer-modified-p t)` instead but they didn't work.

Again, feels like a _me_ problem but if you have time I'd appreciate some pointers on what to look for.

**Comment by douo** [comment]
This sounds a bit strange, but I couldn’t reproduce it in my environment. Also, the BUFFER name is `COMMIT_EDITMSG`, not `COMMIT_MSG` in my environment.

**Comment by martinbaillie** [comment]
> the BUFFER name is COMMIT_EDITMSG, not COMMIT_MSG in my environment

Ah yes, mine too. I didn't pay enough attention when writing this issue.

Sounds like a _me_ problem then. I could only make it work with some advising:

```elisp
(advice-add 'magit-gptcommit-commit-accept :after
            (lambda ()
              (when-let ((buf (magit-commit-message-buffer)))
                (with-current-buffer buf
                  (save-buffer)))))
```

I notice if I run the magitgpt-accept function _again_ whilst in the COMMIT_EDITMSG buffer (so already after doing <kbd>c</kbd> then <kbd>C</kbd> from the `magit-status` transient menu) then it works. Would you expect the user to have to do it twice?

I can reproduce with this slimline config:

```elisp
(require 'use-package)
(setq auth-sources '((:source "REDACTED")))
(use-package magit)
(use-package dash)
(use-package llm :custom (llm-warn-on-nonfree nil))
(use-package magit-gptcommit
  :demand t
  :after magit
  :init
  (require 'llm-openai)

  :custom
  (magit-gptcommit-llm-provider
   (make-llm-openai
    :key (auth-source-pick-first-password :host "api.openai.com" :user "REDACTED")
    :chat-model "gpt-4o"))

  :config
  (magit-gptcommit-status-buffer-setup)
  (magit-gptcommit-mode 1))
```

**Comment by osheari1** [comment]
I've been using this package for quite a while, and around the same time this was posted, I started getting the same error message.

**Comment by osheari1** [comment]
But it looks like it fixed itself after upgrading now. (I'm using doom emacs).

**Comment by alexjp** [comment]
I just found out about magit-gptcommit, tried it and was having the same issue.

I noticed that the edit message buffer had to become dirty and then saved, to not have the "empty commit message" error.

The `advice-add` workaround from @martinbaillie worked for me.

**Comment by osheari1** [comment]
Yeah, the advice add works for me as well when using `c C`. However, `c Q` still returns an error.

**Comment by douo** [comment]
After a series of package upgrades in Emacs, I encountered this issue as well. Perhaps it was caused by a behavior change in magit? The reason is that modifications to the `COMMIT_EDITMSG` buffer in the `git-commit-setup-hook` are no longer automatically saved to the file. It now requires a manual save.

**Comment by avishefi** [comment]
I am debugging this now, seems like `magit`'s `git-commit` marks the commit message buffer as not modified using `(set-buffer-modified-p nil)` ([see here](https://github.com/magit/magit/blob/085baa2ac96df90bdb2a1bedd3598df19919d731/lisp/git-commit.el#L610)) and that prevents `with-editor` from [writing the changes](https://github.com/magit/with-editor/blob/ca902ae02972bdd6919a902be2593d8cb6bd991b/lisp/with-editor.el#L390) even though it invokes `save-buffer` (which saves only if the buffer is modified).

These are not the actual cause since they exist for some time now, but it seems it is related to some new behavior in `magit`.
